### DISCUSSION TITLE
Is it a bug?

### DISCUSSION THREAD

**Discussion by Anmol-Bakshi** [neperfepx/neper#871] [Q&A] [answered]
neper -T -n 2000 -morpho gg -morphooptistop "val=1e-1" -domain "cube(5,2,30)" -transform "cut(cube(-10,1.5,-10,10,7.06,22.94,2),cube(3.5,10,-10,10,7.06,22.94,2),cylinder(3.5,0,15,0,1,0,0.5),cylinder(1.5,0,15,0,1,0,0.5))" -reg 1

 neper -M n2000-id1.tess -statelset vol -rcl 0.5 -order 1 -interface continuous -format inp -o tensile-25-notch -format msh

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 48 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/ffg/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] n2000-id1.tess -statelset vol -rcl 0.5 -order 1
         -interface continuous -format inp -o tensile-25-notch -format
         msh
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `n2000-id1.tess'...
Info   :     [i] Parsed file `n2000-id1.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.1335) 100%
Info   :   - 0D meshing... 100%
Info   :   - 1D meshing... 100%
Info   :   - 2D meshing...  12% (0.37|0.89/71%|14%|15%)bnodeqty = 14 (*pN).NodeQty = 7
Error  : You have discovered a bug in Neper!  Please file an issue at
         https://github.com/rquey/neper/issues.  Thank you.

**Comment by rquey** [comment, marked as answer]
Not really. Try to specify a different `-id` value (not 1) in your `-T` command.

**Reply by Anmol-Bakshi** [reply]
As asked by you, I have changed -ids and tried, every time I am getting different error or a bug. Can you suggest/guide what can be done to remove them.
I am getting like this for six cases:
(1)  neper -T -n 2000 -morpho gg -morphooptistop "val=1e-1" -domain "cube(5,2,30)" -transform "cut(cube(-10,1.5,-10,10,7.06,22.94,2),cube(3.5,10,-10,10,7.06,22.94,2),cylinder(3.5,0,15,0,1,0,0.5),cylinder(1.5,0,15,0,1,0,0.5))" -id 2 -reg 1

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 48 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/ffg/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 2000 -morpho gg -morphooptistop val=1e-1 -domain
         cube(5,2,30) -transform
         cut(cube(-10,1.5,-10,10,7.06,22.94,2),cube(3.5,10,-10,10,7.06,22.94,2),cylinder(3.5,0,15,0,1,0,0.5),cylinder(1.5,0,15,0,1,0,0.5))
         -id 2 -reg 1
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :   - Setting seeds... 100%
Info   :   - Running tessellation...
Info   :     > Initial solution: f   =1.319673114
Info   :     > Iteration  34206: fmin=0.099998364 f=0.099998364
Info   :     > Final solution  : f   =0.099998364 (34206 iterations)
Info   :     > Reached `val' criterion.
Info   : Generating crystal orientations...
Info   : Transforming tessellation...
Info   :   - Cutting (experimental)...
Info   :     > cube(-10,1.5,-10,10,7.06,22.94,2)...
Info   :     > cube(3.5,10,-10,10,7.06,22.94,2)...
Info   :     > cylinder(3.5,0,15,0,1,0,0.5)...
Info   :     > cylinder(1.5,0,15,0,1,0,0.5)...
Info   : Regularizing tessellation...
Info   :   - loop 2/2: 100% del=2466
Info   : Writing results...
Info   :     [o] Writing file `n2000-id2.tess'...
Info   :     [o] Wrote file `n2000-id2.tess'.
Info   : Elapsed time: 87.455 secs.
========================================================================

ffg@ffg:~/Anmol/PhD/Neper/tensile-sample-S6$ neper -M n2000-id2.tess -statelset vol -rcl 0.5 -order 1 -interface continuous -format inp -o tensile-25-notch -format msh                                                           
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 48 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/ffg/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] n2000-id2.tess -statelset vol -rcl 0.5 -order 1
         -interface continuous -format inp -o tensile-25-notch -format
         msh
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `n2000-id2.tess'...
Info   :     [i] Parsed file `n2000-id2.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.1324) 100%
Error  :   - Failed to project 3 faces: 4306 4310 8060.
Info   :     [o] Writing file `failed_faces'...
Info   :     [o] Wrote file `failed_faces'.
Info   :     [o] Writing file `failed_polys'...
Info   :     [o] Wrote file `failed_polys'.
Aborted (core dumped)


(2) neper -T -n 2000 -morpho gg -morphooptistop "val=1e-1" -domain "cube(5,2,30)" -transform "cut(cube(-10,1.5,-10,10,7.06,22.94,2),cube(3.5,10,-10,10,7.06,22.94,2),cylinder(3.5,0,15,0,1,0,0.5),cylinder(1.5,0,15,0,1,0,0.5))" -id 5 -reg 1

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 48 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/ffg/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 2000 -morpho gg -morphooptistop val=1e-1 -domain
         cube(5,2,30) -transform
         cut(cube(-10,1.5,-10,10,7.06,22.94,2),cube(3.5,10,-10,10,7.06,22.94,2),cylinder(3.5,0,15,0,1,0,0.5),cylinder(1.5,0,15,0,1,0,0.5))
         -id 5 -reg 1
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :   - Setting seeds... 100%
Info   :   - Running tessellation...
Info   :     > Initial solution: f   =1.655873613
Info   :     > Iteration  34748: fmin=0.099997660 f=0.099997660
Info   :     > Final solution  : f   =0.099997660 (34748 iterations)
Info   :     > Reached `val' criterion.
Info   : Generating crystal orientations...
Info   : Transforming tessellation...
Info   :   - Cutting (experimental)...
Info   :     > cube(-10,1.5,-10,10,7.06,22.94,2)...
Info   :     > cube(3.5,10,-10,10,7.06,22.94,2)...
Info   :     > cylinder(3.5,0,15,0,1,0,0.5)...
Info   :     > cylinder(1.5,0,15,0,1,0,0.5)...
Info   : Regularizing tessellation...
Info   :   - loop 2/2: 100% del=2359
Info   : Writing results...
Info   :     [o] Writing file `n2000-id5.tess'...
Info   :     [o] Wrote file `n2000-id5.tess'.
Info   : Elapsed time: 92.030 secs.
========================================================================

ffg@ffg:~/Anmol/PhD/Neper/tensile-sample-S6$ neper -M n2000-id5.tess -statelset vol -rcl 0.4 -order 1 -interface continuous -format inp -o tensile-25-notch -format msh                                                           
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 48 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/ffg/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] n2000-id5.tess -statelset vol -rcl 0.4 -order 1
         -interface continuous -format inp -o tensile-25-notch -format
         msh
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `n2000-id5.tess'...
Info   :     [i] Parsed file `n2000-id5.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.1068) 100%
Info   :   - 0D meshing... 100%
Info   :   - 1D meshing... 100%
Info   :   - 2D meshing...   1% (0.74|0.9/83%|10%| 7%)bnodeqty = 19 (*pN).NodeQty = 18
Error  : You have discovered a bug in Neper!  Please file an issue at
         https://github.com/rquey/neper/issues.  Thank you.

Aborted (core dumped)


(3)  neper -T -n 2000 -morpho gg -morphooptistop "val=1e-1" -domain "cube(5,2,30)" -transform "cut(cube(-10,1.5,-10,10,7.06,22.94,2),cube(3.5,10,-10,10,7.06,22.94,2),cylinder(3.5,0,15,0,1,0,0.5),cylinder(1.5,0,15,0,1,0,0.5))" -id 7 -reg 1

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 48 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/ffg/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 2000 -morpho gg -morphooptistop val=1e-1 -domain
         cube(5,2,30) -transform
         cut(cube(-10,1.5,-10,10,7.06,22.94,2),cube(3.5,10,-10,10,7.06,22.94,2),cylinder(3.5,0,15,0,1,0,0.5),cylinder(1.5,0,15,0,1,0,0.5))
         -id 7 -reg 1
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :   - Setting seeds... 100%
Info   :   - Running tessellation...
Info   :     > Initial solution: f   =1.382516102
Info   :     > Iteration  31841: fmin=0.099920538 f=0.099920538
Info   :     > Final solution  : f   =0.099920538 (31841 iterations)
Info   :     > Reached `val' criterion.
Info   : Generating crystal orientations...
Info   : Transforming tessellation...
Info   :   - Cutting (experimental)...
Info   :     > cube(-10,1.5,-10,10,7.06,22.94,2)...
Info   :     > cube(3.5,10,-10,10,7.06,22.94,2)...
Info   :     > cylinder(3.5,0,15,0,1,0,0.5)...
Info   :     > cylinder(1.5,0,15,0,1,0,0.5)...
Info   : Regularizing tessellation...
Info   :   - loop 2/2: 100% del=2442
Info   : Writing results...
Info   :     [o] Writing file `n2000-id7.tess'...
Info   :     [o] Wrote file `n2000-id7.tess'.
Info   : Elapsed time: 86.518 secs.
========================================================================

ffg@ffg:~/Anmol/PhD/Neper/tensile-sample-S6$ neper -M n2000-id7.tess -statelset vol -rcl 0.4 -order 1 -interface continuous -format inp -o tensile-25-notch -format msh                                                           
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 48 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/ffg/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] n2000-id7.tess -statelset vol -rcl 0.4 -order 1
         -interface continuous -format inp -o tensile-25-notch -format
         msh
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `n2000-id7.tess'...
Info   :     [i] Parsed file `n2000-id7.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.1058) 100%
Error  :   - Failed to project 1 face: 7091.
Info   :     [o] Writing file `failed_faces'...
Info   :     [o] Wrote file `failed_faces'.
Info   :     [o] Writing file `failed_polys'...
Info   :     [o] Wrote file `failed_polys'.
Aborted (core dumped)


(4) neper -T -n 2000 -morpho gg -morphooptistop "val=1e-1" -domain "cube(5,2,30)" -transform "cut(cube(-10,1.5,-10,10,7.06,22.94,2),cube(3.5,10,-10,10,7.06,22.94,2),cylinder(3.5,0,15,0,1,0,0.5),cylinder(1.5,0,15,0,1,0,0.5))" -id 17 -reg 1

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 48 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/ffg/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 2000 -morpho gg -morphooptistop val=1e-1 -domain
         cube(5,2,30) -transform
         cut(cube(-10,1.5,-10,10,7.06,22.94,2),cube(3.5,10,-10,10,7.06,22.94,2),cylinder(3.5,0,15,0,1,0,0.5),cylinder(1.5,0,15,0,1,0,0.5))
         -id 17 -reg 1
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :   - Setting seeds... 100%
Info   :   - Running tessellation...
Info   :     > Initial solution: f   =1.453936051
Info   :     > Iteration  33024: fmin=0.099996851 f=0.099996851
Info   :     > Final solution  : f   =0.099996851 (33024 iterations)
Info   :     > Reached `val' criterion.
Info   : Generating crystal orientations...
Info   : Transforming tessellation...
Info   :   - Cutting (experimental)...
Info   :     > cube(-10,1.5,-10,10,7.06,22.94,2)...
Info   :     > cube(3.5,10,-10,10,7.06,22.94,2)...
Info   :     > cylinder(3.5,0,15,0,1,0,0.5)...
Info   :     > cylinder(1.5,0,15,0,1,0,0.5)...
Info   : Regularizing tessellation...
Info   :   - loop 2/2: 100% del=2506
Info   : Writing results...
Info   :     [o] Writing file `n2000-id17.tess'...
Info   :     [o] Wrote file `n2000-id17.tess'.
Info   : Elapsed time: 88.228 secs.
========================================================================

ffg@ffg:~/Anmol/PhD/Neper/tensile-sample-S6$ neper -M n2000-id17.tess -statelset vol -rcl 0.4 -order 1 -interface continuous -format inp -o tensile-25-notch -format msh                                                          
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 48 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/ffg/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] n2000-id17.tess -statelset vol -rcl 0.4 -order 1
         -interface continuous -format inp -o tensile-25-notch -format
         msh
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `n2000-id17.tess'...
Info   :     [i] Parsed file `n2000-id17.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.1058) 100%
Info   :   - 0D meshing... 100%
Info   :   - 1D meshing... 100%
Info   :   - 2D meshing...  14% (0.29|0.9/72%|14%|14%)bnodeqty = 28 (*pN).NodeQty = 7
Error  : You have discovered a bug in Neper!  Please file an issue at
         https://github.com/rquey/neper/issues.  Thank you.

Aborted (core dumped)

(5) neper -T -n 4000 -morpho gg -morphooptistop "val=1e-1" -domain "cube(5,2,30)" -transform "cut(cube(-10,1.5,-10,10,7.06,22.94,2),cube(3.5,10,-10,10,7.06,22.94,2),cylinder(3.5,0,15,0,1,0,0.5),cylinder(1.5,0,15,0,1,0,0.5))" -reg 1

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 48 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/ffg/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 4000 -morpho gg -morphooptistop val=1e-1 -domain
         cube(5,2,30) -transform
         cut(cube(-10,1.5,-10,10,7.06,22.94,2),cube(3.5,10,-10,10,7.06,22.94,2),cylinder(3.5,0,15,0,1,0,0.5),cylinder(1.5,0,15,0,1,0,0.5))
         -reg 1
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :   - Setting seeds... 100%
Info   :   - Running tessellation...
Info   :     > Initial solution: f   =1.078652217
Info   :     > Iteration  59565: fmin=0.099962344 f=0.099962344
Info   :     > Final solution  : f   =0.099962344 (59565 iterations)
Info   :     > Reached `val' criterion.
Info   : Generating crystal orientations...
Info   : Transforming tessellation...
Info   :   - Cutting (experimental)...
Info   :     > cube(-10,1.5,-10,10,7.06,22.94,2)...
Info   :     > cube(3.5,10,-10,10,7.06,22.94,2)...
Info   :     > cylinder(3.5,0,15,0,1,0,0.5)...
Info   :     > cylinder(1.5,0,15,0,1,0,0.5)...
Info   : Regularizing tessellation...
Info   :   - loop 2/2: 100% del=5301
Info   : Writing results...
Info   :     [o] Writing file `n4000-id1.tess'...
Info   :     [o] Wrote file `n4000-id1.tess'.
Info   : Elapsed time: 185.729 secs.
========================================================================

ffg@ffg:~/Anmol/PhD/Neper/tensile-sample-S6$ neper -M n4000-id1.tess -statelset vol -rcl 0.4 -order 1 -interface continuous -format inp -o tensile-25-notch -format msh                                                           
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 48 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/ffg/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] n4000-id1.tess -statelset vol -rcl 0.4 -order 1
         -interface continuous -format inp -o tensile-25-notch -format
         msh
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `n4000-id1.tess'...
Info   :     [i] Parsed file `n4000-id1.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.08485) 100%
Info   :   - 0D meshing... 100%
Info   :   - 1D meshing... 100%
Info   :   - 2D meshing...   6% (0.67|0.9/75%|13%|12%)bnodeqty = 24 (*pN).NodeQty = 13
Error  : You have discovered a bug in Neper!  Please file an issue at
         https://github.com/rquey/neper/issues.  Thank you.

Aborted (core dumped)

(6) neper -T -n 4000 -morpho gg -morphooptistop "val=1e-1" -domain "cube(5,2,30)" -transform "cut(cube(-10,1.5,-10,10,7.06,22.94,2),cube(3.5,10,-10,10,7.06,22.94,2),cylinder(3.5,0,15,0,1,0,0.5),cylinder(1.5,0,15,0,1,0,0.5))" -id 25 -reg 1

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 48 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/ffg/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 4000 -morpho gg -morphooptistop val=1e-1 -domain
         cube(5,2,30) -transform
         cut(cube(-10,1.5,-10,10,7.06,22.94,2),cube(3.5,10,-10,10,7.06,22.94,2),cylinder(3.5,0,15,0,1,0,0.5),cylinder(1.5,0,15,0,1,0,0.5))
         -id 25 -reg 1
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :   - Setting seeds... 100%
Info   :   - Running tessellation...
Info   :     > Initial solution: f   =1.113492142
Info   :     > Iteration  58298: fmin=0.099994444 f=0.099994444
Info   :     > Final solution  : f   =0.099994444 (58298 iterations)
Info   :     > Reached `val' criterion.
Info   : Generating crystal orientations...
Info   : Transforming tessellation...
Info   :   - Cutting (experimental)...
Info   :     > cube(-10,1.5,-10,10,7.06,22.94,2)...
Info   :     > cube(3.5,10,-10,10,7.06,22.94,2)...
Info   :     > cylinder(3.5,0,15,0,1,0,0.5)...
Info   :     > cylinder(1.5,0,15,0,1,0,0.5)...
Info   : Regularizing tessellation...
Info   :   - loop 2/2: 100% del=5397
Info   : Writing results...
Info   :     [o] Writing file `n4000-id25.tess'...
Info   :     [o] Wrote file `n4000-id25.tess'.
Info   : Elapsed time: 178.492 secs.
========================================================================

ffg@ffg:~/Anmol/PhD/Neper/tensile-sample-S6$ neper -M n4000-id25.tess -statelset vol -rcl 0.4 -order 1 -interface continuous -format inp -o tensile-25-notch -format msh                                                          
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 48 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/ffg/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] n4000-id25.tess -statelset vol -rcl 0.4 -order 1
         -interface continuous -format inp -o tensile-25-notch -format
         msh
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `n4000-id25.tess'...
Info   :     [i] Parsed file `n4000-id25.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.08435) 100%
Info   :   - 0D meshing... 100%
Info   :   - 1D meshing... 100%
Info   :   - 2D meshing...   4% (0.11|0.9/75%|13%|12%)
*pbnodeqty = 29
nodeqty1d = 34
min_dist_max = 0.000000
Aborted (core dumped)

**Comment by anujyadav0693** [comment]
Anmol, how do you install the latest version 4.9.1-1? in GitHub latest is showing 4.9.0
please reply.
thank you

**Comment by Anmol-Bakshi** [comment]
it is there, u can also see in screenshot.
![Screenshot 2024-07-02 152502](https://github.com/neperfepx/neper/assets/108575045/03424b89-df39-4ef7-a6ad-b52aac35776f)

**Comment by anujyadav0693** [comment]
Thank you brother but when I going to install .tar.gz file. From where I
get this latest file because in GitHub page latest .tar.gz file is for
4.9.0 thankyou

**Comment by Anmol-Bakshi** [comment]
i think u can again download the src folder and install.

**Comment by anujyadav0693** [comment]
Thank you brother i found

**Comment by Anmol-Bakshi** [comment]
then i am not sure, maybe it has been removed or something, but i followed the same procedure: downloaded the src file, extracted it and using commands installed it.

**Comment by anujyadav0693** [comment]
I also do the same it is working
